### PR TITLE
Admin Create/edit car, assign/unassign User 

### PR DESCRIPTION
**Pull request by mahdibenromdhane**

### CODE REVIEW

File: src/types/Authorization.ts [hunk 1]
```diff
@@ -69,6 +69,8 @@ export enum Entity {
   CAR_CATALOGS = 'CarCatalogs',
   CAR = 'Car',
   CARS = 'Cars',
+  USER_CAR='UserCar',
+  USERS_CARS='usersCars',
   INVOICE = 'Invoice',
```
Comment (jerome-benoit) [inline src/types/Authorization.ts:74]: Case issue.

File: src/types/Authorization.ts [hunk 2]
```diff
@@ -69,6 +69,8 @@ export enum Entity {
   CAR_CATALOGS = 'CarCatalogs',
   CAR = 'Car',
   CARS = 'Cars',
+  USER_CAR='UserCar',
+  USERS_CARS='usersCars',
```
Comment (jerome-benoit) [inline src/types/Authorization.ts:73]: Case issue.

File: src/storage/mongodb/CarStorage.ts [hunk 3]
```diff
@@ -662,4 +759,119 @@ export default class CarStorage {
...
+    const filters: any = {};
+    if (params.active) {
+      filters.active = params.active;
+    }
+    // Limit on Car for Basic Users
+    if (params.userscarsIDs && params.userscarsIDs.length > 0) {
+      // Build filter
+      filters._id = { $in: params.userscarsIDs.map((userscarsID) => Utils.convertToObjectID(userscarsID)) };
+    }
+    if (params.carIDs && params.carIDs.length > 0) {
```
Comment (jerome-benoit) [inline src/storage/mongodb/CarStorage.ts:795]: In Utils, you have an helper function to test for empty array.

**Comment by OliveGerste** [comment]
Hi @mahdibenromdhane 

I tested again and everything seems to work properly. The only thing I recognized is, that you can assign users to cars, which does not belong to you via postman (basic auth). 
And also create a pool car via postman with basic auth.
When these issues are fixed you can move both PRs in the to be reviewed column.

Thanks and best regards,
Tobias

**Comment by mahdibenromdhane** [comment]
> Hi @mahdibenromdhane
> 
> I tested again and everything seems to work properly. The only thing I recognized is, that you can assign users to cars, which does not belong to you via postman (basic auth).
> And also create a pool car via postman with basic auth.
> When these issues are fixed you can move both PRs in the to be reviewed column.
> 
> Thanks and best regards,
> Tobias

Yes, you're right ! I will fix it , thank you

**Pull request merged**